Add 'discounted' field to 'Price' table

choogland opened the pull request:
- Add 'discounted' (bool) field to 'Price' table
- Add migration script
- Update mermaid diagram & SVG diagram
- Fix missing import in migration script d483c555d6d1_add_field_origin_to_source_table.py

SkypLabs commented:
Your branch doesn't contain any new commits at the moment. Also, your PR needs to target the `develop` branch, not `main`.

Commit:
Add 'discounted' field to 'Price' table

- Add 'discounted' (bool) field to 'Price' table
- Add migration script
- Update mermaid diagram & SVG diagram

SkypLabs reviewed (commented):
You seem to have imported unused classes. Is there any reason for this?

Otherwise, LGTM.

choogland commented:
I got an error when running alembic revision:
sqlalchemy.exc.NoReferencedTableError: Foreign key associated with column 'source.product_id' could not find table 'product' with which to generate a foreign key to target column 'id'

SkypLabs commented:
> I got an error when running alembic revision: sqlalchemy.exc.NoReferencedTableError: Foreign key associated with column 'source.product_id' could not find table 'product' with which to generate a foreign key to target column 'id'

I think the problem comes from something else. I can go all the way back to the orignal revision and forth with Alembic, and I don't see any problem:

```
proteines-resilientes on  feature/env-facts [$!+?] is 📦 v0.1.0 via  v3.13.7 (proteines-resilientes-py3.13) 
❯ alembic downgrade base
INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running downgrade d483c555d6d1 -> 74f8c9658d25, Add field 'origin' to 'Source' table
INFO  [alembic.runtime.migration] Running downgrade 74f8c9658d25 -> , Add 'disabled' field to 'Product' table

proteines-resilientes on  feature/env-facts [$!+?] is 📦 v0.1.0 via  v3.13.7 (proteines-resilientes-py3.13) 
❯ alembic upgrade head
INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade  -> 74f8c9658d25, Add 'disabled' field to 'Product' table
INFO  [alembic.runtime.migration] Running upgrade 74f8c9658d25 -> d483c555d6d1, Add field 'origin' to 'Source' tabl
```

Commit:
Remove unecessary import

choogland commented:
OK. I don't know why. Could it be because I'm on Windows, althought I'm using WSL ubuntu 24.04 ?
I have removed the 2 unecessary imports. But apparently the tests are still failing.

SkypLabs commented:
> OK. I don't know why. Could it be because I'm on Windows, althought I'm using WSL ubuntu 24.04 ? I have removed the 2 unecessary imports. But apparently the tests are still failing.

I don't think it's Windows-related. Can you run `poetry install` again after making sure to use the latest commit on `develop`, and then, run the tests one more time?

In all cases, thanks for having removed your fix from this PR, but I invite you to create an issue to investigate this weird behaviour.

SkypLabs reviewed (changes requested):
Please run `ruff format` to fix the code formatting, and then it will be good for a merge.

SkypLabs commented:
FYI, I just created an issue (#51) to run the migration scripts on the CI.

Commit:
Fix code formatting

Run ruff format to fix formatting

SkypLabs reviewed (approved):
I moved your commits related to detecting the discounted prices on Auchan into a separate branch as it wasn't the purpose of this PR. It wasn't a big deal, but it prevented me from squashing your commits all together.
